core/src/main/scala/org/apache/spark/scheduler/TaskSetManager.scala

```diff
@@ -469,8 +480,14 @@ private[spark] class TaskSetManager(
           extraResources,
           serializedTask)
       }
+      val hasPendingTasks = pendingTasks.all.nonEmpty || pendingSpeculatableTasks.all.nonEmpty
```

Comment:
@tgravescs 
It turns out that AFAICT there is no case that would hit empty task description, ANY locality, no pending tasks, yet there are pending speculative tasks, because once there are no regular pending tasks, the speculative tasks don't follow delay scheduling (you'd only get empty task if there were none pending). The locality level is only determined by non speculative pending tasks. See [here](https://github.com/apache/spark/blob/master/core/src/main/scala/org/apache/spark/scheduler/TaskSetManager.scala#L543-L546) 

I ended up not adding a test because it was complex and ugly (given it isn't currently possible).
It required overriding internal fields of TSM in odd ways.
Think it is better to add an ugly test, take out the check on speculative tasks, or leave as is?